app/views/posts/index.html.erb

```diff
@@ -1 +1,36 @@
-<h1>Here is a list of posts for a given user</h1>
\ No newline at end of file
+<section class= "users-card">
+  <div class="user-profile" >
+     <img src="<%=@user.photo%>">
+    <div>
+    <h1> <%=@user.name%></h1>
+    <h3> Number of posts: <%=@user.postsCounter%><h3>
+    </div>
+  <div>
+   <ul>
+   <% @posts.each do |post|%>
+     <li>
+     <a href="<%= user_post_path(@user, post) %>">
+      <div>
+          <h2><%=post.title%><h2>
+          <h3><%=post.text%></h3>
+          <p>Comments:<%=post.comments_counter%><p>
+          <p>Likes:<%=post.likes_counter%></p>
+       <div>
```

Comment:
- [OPTIONAL] Great job so far 👏🏻  However You may use Partial templates eg:  `app/views/users/_user.html.erb`, as a better practice instead of using `@objects.each`. Partial templates (partials) are a way of breaking the rendering process into more manageable chunks. Partials allow you to extract pieces of code from your templates to separate files and also reuse them throughout your templates. For more documentation visit this [link](https://riptutorial.com/ruby-on-rails/example/2875/partials)



```ruby
    # app/views/users/_user.html.erb

     #   HTML Code Here displaying
     #   user.photo, user.name and user.posts_counter

    # app/views/users/index.html.erb

    <%= render @users %>
```

Reply:
Thanks @BenMukebo , I finally got what was causing the error!,
Thanks also for suggesting me to partial templates , I have realised that they make code more DRY,